qiskit/providers/basic_provider/basic_simulator.py

```diff
@@ -555,55 +458,52 @@ def run(
                 )
             else:
                 out_options[key] = value
-        qobj = _assemble(run_input, self, **out_options)
-        qobj_options = qobj.config
-        self._set_options(qobj_config=qobj_options, backend_options=backend_options)
+        self._set_options(backend_options=backend_options)
         job_id = str(uuid.uuid4())
-        job = BasicProviderJob(self, job_id, self._run_job(job_id, qobj))
+        job = BasicProviderJob(self, job_id, self._run_job(job_id, run_input))
         return job
 
-    def _run_job(self, job_id: str, qobj: QasmQobj) -> Result:
-        """Run experiments in qobj
+    def _run_job(self, job_id: str, run_input) -> Result:
+        """Run circuits in job.
 
         Args:
             job_id: unique id for the job.
-            qobj: job description
+            run_input: circuits to be run.
 
         Returns:
             Result object
         """
-        self._validate(qobj)
+        if isinstance(run_input, QuantumCircuit):
+            run_input = [run_input]
+
+        self._validate(run_input)
         result_list = []
-        self._shots = qobj.config.shots
-        self._memory = getattr(qobj.config, "memory", False)
-        self._qobj_config = qobj.config
+        self._memory = True
```

Comment:
Why is this set to True rather than using the value given in `self.options`? 
Another thing BTW worth testing in `test_basic_simulator.py` (that `memory` is honored)